index.js

```diff
@@ -313,16 +313,14 @@ function objEquiv(a, b, opts, channel) {
   var aIsBuffer = isBuffer(a);
   var bIsBuffer = isBuffer(b);
   if (aIsBuffer !== bIsBuffer) { return false; }
-  if (aIsBuffer || bIsBuffer) { // && would work too, because both are true or both false here
+  if (aIsBuffer) {
     if (a.length !== b.length) { return false; }
     for (i = 0; i < a.length; i++) {
       if (a[i] !== b[i]) { return false; }
     }
     return true;
   }
 
-  if (typeof a !== typeof b) { return false; }
-
```

Comment:
This is already checked at the top of the function:
https://github.com/inspect-js/node-deep-equal/blob/86a159b72c3e3620285498113251220d3c5d05f0/index.js#L275